File: frontendcapstone/src/app/Group.ts
export class Group {
    id: number;
    name: string;
    announcementMessages: Message[];
    discussionMessages: Message[];
    admins: number[];
    users: number[];
    tags: string[];

    constructor(
        id: number,
        name: string,
        announcementMessages: Message[] = [],
        discussionMessages: Message[] = [],
        admins: number[] = [],
        users: number[] = [],
        tags: string[] = []
    ) {
        this.id = id;
        this.name = name;
        this.announcementMessages = announcementMessages;
        this.discussionMessages = discussionMessages;
        this.admins = admins;
        this.users = users;
        this.tags = tags;
    }

    toString(): string {
        return `Group {
            id: ${this.id},
            name: "${this.name}",
            admins: [${this.admins.join(', ')}],
            users: [${this.users.join(', ')}],
            tags: [${this.tags.join(', ')}],
            announcementMessages: [${this.announcementMessages.map(m => m.toString()).join(', ')}],
            discussionMessages: [${this.discussionMessages.map(m => m.toString()).join(', ')}]
        }`;
    }
}

export class Message {
    id?: number;
    content?: string;
    timestamp?: string;
    sentBy?: number;

    constructor(id?: number, content?: string, timestamp?: string, sentBy?: number) {
        this.id = id;
        this.content = content;
        this.timestamp = timestamp;
        this.sentBy = sentBy;
    }

    toString(): string {
        return `Message {
            id: ${this.id},
            content: "${this.content}",
            timestamp: "${this.timestamp}",
            sentBy: ${this.sentBy}
        }`;
    }
}

File: LivingSyncbackend/announcement/src/main/java/com/livingsync/annoucements/model/Group.java
package com.livingsync.annoucements.model;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "chat_group")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Group {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
    @JoinColumn(name = "group_id")
    private List<Message> announcementMessages = new ArrayList<>();

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
    @JoinColumn(name = "group_id")
    private List<Message> discussionMessages = new ArrayList<>();

    @ElementCollection
    @CollectionTable(name = "group_admins", joinColumns = @JoinColumn(name = "group_id"))
    @Column(name = "admin_id")
    private List<Long> admins = new ArrayList<>();

    @ElementCollection
    @CollectionTable(name = "group_users", joinColumns = @JoinColumn(name = "group_id"))
    @Column(name = "user_id")
    private List<Long> users = new ArrayList<>();

    @ElementCollection
    @CollectionTable(name = "group_tags", joinColumns = @JoinColumn(name = "group_id"))
    @Column(name = "tag")
    private List<String> tags = new ArrayList<>();

    // Getters and setters
}
